File: Kotlin/asciiart.kt
// asciiart Kotlin
fun main() {
    for (y in -12..12) {
        for (x in -39..39) {
            val ca = x * 0.0458
            val cb = y * 0.08333
            var a = ca
            var b = cb
            var s = " "
            for (i in 0..15) {
                val t = a * a - b * b + ca
                b = 2 * a * b + cb
                a = t
                if ((a * a + b * b) > 4) {
                    s = "%X".format(i)
                    break
                }
            }
            print(s)
        }
        println()
    }
}

File: Clojure/asciiart.clj
; asciiart Clojure

(defn mandel [i a b ca cb]
    (if (= 16 i)
        (print " ")
        (let [an (+ (- (* a a) (* b b)) ca)
              bn (+ (* 2 a b) cb)]
            (if (> (+ (* an an) (* bn bn)) 4)
                (print (format "%X" i))
                (recur (inc i) an bn ca cb)))))

(doseq [y (range -12 13)]
    (doseq [x (range -39 40)]
        (let [ca (* x 0.0458)
              cb (* y 0.08333)]
            (mandel 0 ca cb ca cb)))
    (println))
